<launch>
	<arg name="side_length" default="2" />
	<arg name="rotation" default="1" />
	<arg name="speed" default="1" />
	<arg name="radius" default="2" />
	
	<arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>

	<param name="model" value="$(arg model)"/>

	<include file="$(find turtlebot3_bringup)/launch/includes/description.launch.xml">
		<arg name="model" value="$(arg model)" />
	</include>

	<node pkg="turtlebot3_fake" type="turtlebot3_fake_node" name="turtlebot3_fake_node" output="screen" />

	<node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
		<param name="publish_frequency" type="double" value="50.0" />
	</node>

	<node name="draw_sqare_server" pkg = "turtlebot3_practice_move" type="draw_square_server" output = "screen" respawn="true" />
	<node name="draw_sqare_client" pkg = "turtlebot3_practice_move" type="draw_square_client" output = "screen" />
	
	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find turtlebot3_fake)/rviz/turtlebot3_fake.rviz"/>
</launch>